flink-core/src/test/java/org/apache/flink/api/java/typeutils/FieldAccessorTest.java

```diff
@@ -0,0 +1,280 @@
+				(TupleTypeInfo<Tuple2<String, Integer>>) TypeExtractor.getForObject(t);
+
+		FieldAccessor<Tuple2<String, Integer>, String> f0 = tpeInfo.getFieldAccessor("f0", null);
+		assertEquals("aa", f0.get(t));
+		assertEquals("aa", t.f0);
+		t = f0.set(t, "b");
+		assertEquals("b", f0.get(t));
+		assertEquals("b", t.f0);
+
+		FieldAccessor<Tuple2<String, Integer>, Integer> f1 = tpeInfo.getFieldAccessor("f1", null);
+		assertEquals(5, (int) f1.get(t));
+		assertEquals(5, (int) t.f1);
+		t = f1.set(t, 7);
+		assertEquals(7, (int) f1.get(t));
+		assertEquals(7, (int) t.f1);
+		assertEquals("b", f0.get(t));
+		assertEquals("b", t.f0);
+
+
+		FieldAccessor<Tuple2<String, Integer>, Integer> f1n = tpeInfo.getFieldAccessor(1, null);
+		assertEquals(7, (int) f1n.get(t));
+		assertEquals(7, (int) t.f1);
+		t = f1n.set(t, 10);
+		assertEquals(10, (int) f1n.get(t));
+		assertEquals(10, (int) f1.get(t));
+		assertEquals(10, (int) t.f1);
+		assertEquals("b", f0.get(t));
+		assertEquals("b", t.f0);
+	}
+
+	@Test
+	public void testTupleInTuple() {
+		Tuple2<String, Tuple3<Integer, Long, Double>> t = Tuple2.of("aa", Tuple3.of(5, 9L, 2.0));
+		TupleTypeInfo<Tuple2<String, Tuple3<Integer, Long, Double>>> tpeInfo =
+				(TupleTypeInfo<Tuple2<String, Tuple3<Integer, Long, Double>>>)TypeExtractor.getForObject(t);
+
+		FieldAccessor<Tuple2<String, Tuple3<Integer, Long, Double>>, String> f0 = tpeInfo.getFieldAccessor("f0", null);
+		assertEquals("aa", f0.get(t));
+		assertEquals("aa", t.f0);
+
+		FieldAccessor<Tuple2<String, Tuple3<Integer, Long, Double>>, Double> f1f2 = tpeInfo.getFieldAccessor("f1.f2", null);
+		assertEquals(2.0, f1f2.get(t), 0);
+		assertEquals(2.0, t.f1.f2, 0);
+		t = f1f2.set(t, 3.0);
+		assertEquals(3.0, f1f2.get(t), 0);
+		assertEquals(3.0, t.f1.f2, 0);
+		assertEquals("aa", f0.get(t));
+		assertEquals("aa", t.f0);
+
+		FieldAccessor<Tuple2<String, Tuple3<Integer, Long, Double>>, Tuple3<Integer, Long, Double>> f1 = tpeInfo.getFieldAccessor("f1", null);
+		assertEquals(Tuple3.of(5, 9L, 3.0), f1.get(t));
+		assertEquals(Tuple3.of(5, 9L, 3.0), t.f1);
+		t = f1.set(t, Tuple3.of(8, 12L, 4.0));
+		assertEquals(Tuple3.of(8, 12L, 4.0), f1.get(t));
+		assertEquals(Tuple3.of(8, 12L, 4.0), t.f1);
+		assertEquals("aa", f0.get(t));
+		assertEquals("aa", t.f0);
+
+		FieldAccessor<Tuple2<String, Tuple3<Integer, Long, Double>>, Tuple3<Integer, Long, Double>> f1n = tpeInfo.getFieldAccessor(1, null);
+		assertEquals(Tuple3.of(8, 12L, 4.0), f1n.get(t));
+		assertEquals(Tuple3.of(8, 12L, 4.0), t.f1);
+		t = f1n.set(t, Tuple3.of(10, 13L, 5.0));
+		assertEquals(Tuple3.of(10, 13L, 5.0), f1n.get(t));
+		assertEquals(Tuple3.of(10, 13L, 5.0), f1.get(t));
+		assertEquals(Tuple3.of(10, 13L, 5.0), t.f1);
+		assertEquals("aa", f0.get(t));
+		assertEquals("aa", t.f0);
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void tupleFieldAccessorOutOfBoundsTest() {
+		try {
+			FieldAccessor<Tuple2<Integer, Integer>, Integer> fieldAccessor =
```

Comment:
Ah yes, you are right, removing the cast did it.
39e4d96e686d621ca6609a9f80665862c2a004d3